src/golang/lib/job/k8s.go

```diff
@@ -25,27 +25,42 @@ type k8sJobManager struct {
 	conf      *K8sJobManagerConfig
 }
 
-func NewK8sJobManager(conf *K8sJobManagerConfig) (*k8sJobManager, error) {
-	k8sClient, err := k8s.CreateK8sClient(conf.KubeconfigPath, conf.UseSameCluster)
-	if err != nil {
-		return nil, errors.Wrap(err, "Error while creating K8sClient")
-	}
-
-	err = k8s.CreateNamespaces(k8sClient)
+func setupNamespaceAndSecrets(k8sClient *kubernetes.Clientset, conf *K8sJobManagerConfig) error {
+	err := k8s.CreateNamespaces(k8sClient)
 	if err != nil {
-		return nil, errors.Wrap(err, "Error while creating K8s Namespaces")
+		return errors.Wrap(err, "Error while creating K8s Namespaces")
 	}
 
 	secretsMap := map[string]string{}
 	secretsMap[k8s.AwsAccessKeyIdName] = conf.AwsAccessKeyId
 	secretsMap[k8s.AwsAccessKeyName] = conf.AwsSecretAccessKey
 	err = k8s.CreateSecret(context.TODO(), k8s.AwsCredentialsSecretName, secretsMap, k8sClient)
 	if err != nil {
-		return nil, errors.Wrap(err, "Error while creating K8s Secrets")
+		return errors.Wrap(err, "Error while creating K8s Secrets")
+	}
+
+	return nil
+}
+
+func (j *k8sJobManager) initialize() error {
+	k8sClient, err := k8s.CreateK8sClient(j.conf.KubeconfigPath, j.conf.UseSameCluster)
+	if err != nil {
+		return errors.Wrap(err, "Error while creating K8sClient")
+	}
+
+	err = setupNamespaceAndSecrets(k8sClient, j.conf)
+	if err != nil {
+		return err
 	}
 
+	j.k8sClient = k8sClient
+
+	return nil
+}
+
+func NewK8sJobManager(conf *K8sJobManagerConfig) (*k8sJobManager, error) {
```

Comment:
@likawind @kenxu95 @hsubbaraj-spiral I refactored the job manager implementation based on our discussion.

Reply:
Can you add a comment on the `k8sClient` about why we set it to nil and only initialize in `Poll()` or `Launch()`?